yb-voyager/src/tgtdb/event.go

```diff
@@ -95,12 +146,150 @@ func (event *Event) getDeleteStmt(targetSchema string) string {
 	}
 	var whereClauses []string
 	for column, value := range event.Key {
+		if value == nil { // value can't be nil for keys
+			panic("key value is nil")
+		}
 		whereClauses = append(whereClauses, fmt.Sprintf("%s = %s", column, *value))
 	}
 	whereClause := strings.Join(whereClauses, " AND ")
 	return fmt.Sprintf(deleteTemplate, tableName, whereClause)
 }
 
+func (event *Event) getPreparedInsertStmt(targetSchema string) string {
+	tableName := event.SchemaName + "." + event.TableName
+	if targetSchema != "" {
+		tableName = targetSchema + "." + event.TableName
+	}
+
+	columnList := make([]string, 0, len(event.Fields))
+	valueList := make([]string, 0, len(event.Fields))
+	keys := utils.GetMapKeysSorted(event.Fields)
+	for pos, key := range keys {
+		columnList = append(columnList, key)
+		valueList = append(valueList, fmt.Sprintf("$%d", pos+1))
+	}
+	columns := strings.Join(columnList, ", ")
+	values := strings.Join(valueList, ", ")
+	stmt := fmt.Sprintf(insertTemplate, tableName, columns, values)
+	return stmt
+}
+
+// NOTE: PS for each event of same table can be diffrent as it depends on columns being updated
+func (event *Event) getPreparedUpdateStmt(targetSchema string) string {
+	tableName := event.SchemaName + "." + event.TableName
+	if targetSchema != "" {
+		tableName = targetSchema + "." + event.TableName
+	}
+
+	var setClauses []string
+	keys := utils.GetMapKeysSorted(event.Fields)
+	for pos, key := range keys {
+		setClauses = append(setClauses, fmt.Sprintf("%s = $%d", key, pos+1))
+	}
+	setClause := strings.Join(setClauses, ", ")
+
+	var whereClauses []string
+	keys = utils.GetMapKeysSorted(event.Key)
+	for i, key := range keys {
+		pos := i + 1 + len(event.Fields)
+		whereClauses = append(whereClauses, fmt.Sprintf("%s = $%d", key, pos))
+	}
+	whereClause := strings.Join(whereClauses, " AND ")
+	return fmt.Sprintf(updateTemplate, tableName, setClause, whereClause)
+}
+
+func (event *Event) getPreparedDeleteStmt(targetSchema string) string {
+	tableName := event.SchemaName + "." + event.TableName
+	if targetSchema != "" {
+		tableName = targetSchema + "." + event.TableName
+	}
+	var whereClauses []string
+	keys := utils.GetMapKeysSorted(event.Key)
+	for pos, key := range keys {
+		whereClauses = append(whereClauses, fmt.Sprintf("%s = $%d", key, pos+1))
+	}
+	whereClause := strings.Join(whereClauses, " AND ")
+	return fmt.Sprintf(deleteTemplate, tableName, whereClause)
+}
+
+func (event *Event) getInsertParams() []interface{} {
+	var params []interface{}
+	keys := utils.GetMapKeysSorted(event.Fields)
+	for _, key := range keys {
+		value := event.Fields[key]
+		if value == nil {
+			params = append(params, nil)
+			continue
+		}
+		unquotedValue, err := strconv.Unquote(*value)
+		if err != nil {
+			unquotedValue = *value
+		}
+		params = append(params, unquotedValue)
+	}
```

Comment:
This code structure is repeated four times in the file. Factor out in a separate function e.g. `getMapValuesForQuery()`.